<a href="https://colab.research.google.com/github/OmarHernandezGzz/FCFM-MineriaDeDatos/blob/main/Ej_Limpieza_Equipo7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

* **EQUIPO #7**
* **Grupo:** 001 , Lunes(7:00-10:00)
* **Integrantes:** 
 * 1853256  Trejo Rodriguez Adriana Jackeline
 * 1848627 Villegas Kane Aaron
 * 1866054 Pasillas Moreno Jesus
 * 1855060 Salazar Narvaez Daniela
 * 1850325 Hernandez Gonzalez Omar







**Netflix Movies and TV Shows**: https://www.kaggle.com/shivamb/netflix-shows

In [ ]:
#a) Importar librerias y base de datos
import pandas as pd
import numpy as np
import json
base = pd.read_csv("netflix_titles.csv")

Aqui importamos la base de datos con la que trabajaremos.

In [ ]:
#b) Visualizar los datos
base.shape 

(7787, 12)

Esta funcion nos dice cuantas filas y columnas tiene nuestra base de datos. 


In [ ]:
base.head(3)

show_id  ...                                        description
0      s1  ...  In a future where the elite inhabit an island ...
1      s2  ...  After a devastating earthquake hits Mexico Cit...
2      s3  ...  When an army recruit is found dead, his fellow...

[3 rows x 12 columns]

In [ ]:
base.tail(3)

show_id  ...                                        description
7784   s7785  ...  In this documentary, South African rapper Nast...
7785   s7786  ...  Dessert wizard Adriano Zumbo looks for the nex...
7786   s7787  ...  This documentary delves into the mystique behi...

[3 rows x 12 columns]

Estas funciones te muestras los primeros 3 y los ultimos 3 datos de la base de datos respectivamente.

In [ ]:
#c) Descripcion basica de columnas
base.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description'],
      dtype='object')

Esta funcion nos ayuda a saber el nombre de las columnas de nuestra base de datos.

In [ ]:
base.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

Esta funcion nos muestra el tipo de dato que tienen los elementos de las columnas.

In [ ]:
base.isnull().any() 

show_id         False
type            False
title           False
director         True
cast             True
country          True
date_added       True
release_year    False
rating           True
duration        False
listed_in       False
description     False
dtype: bool

Esta funcion nos indica si en cierta columna hay celdas vacias o datos nulos. True para indicar que hay datos nulos y False para indicar que NO hay datos nulos.

In [ ]:
base.isnull()

show_id   type  title  director  ...  rating  duration  listed_in  description
0       False  False  False      True  ...   False     False      False        False
1       False  False  False     False  ...   False     False      False        False
2       False  False  False     False  ...   False     False      False        False
3       False  False  False     False  ...   False     False      False        False
4       False  False  False     False  ...   False     False      False        False
...       ...    ...    ...       ...  ...     ...       ...        ...          ...
7782    False  False  False     False  ...   False     False      False        False
7783    False  False  False     False  ...   False     False      False        False
7784    False  False  False      True  ...   False     False      False        False
7785    False  False  False      True  ...   False     False      False        False
7786    False  False  False     False  ...   False     False      False        False

[7787 rows x 12 columns]

Esta funcion nos arroja la misma base de datos pero llena de datos boleanos y estos nos indican donde estan las celdas vacias y en donde si hay datos.

In [ ]:
base.isnull().sum()

show_id            0
type               0
title              0
director        2389
cast             718
country          507
date_added        10
release_year       0
rating             7
duration           0
listed_in          0
description        0
dtype: int64

Esta funcion nos ayuda a saber cuantos datos nulos o celdas vacias hay en cada columna de nuestra base.

In [ ]:
del base["director"]
del base["cast"]

La función de arriba nos sirve para eliminar las columnas que no nos van a servir.

In [ ]:
#d)Clasificación de datos
United_States = base[base['country']=='United States']

United_States.to_csv('Solo_United_States.csv', index = False)
United_States

show_id  ...                                        description
3         s4  ...  In a postapocalyptic world, rag-doll robots hi...
4         s5  ...  A brilliant group of students become card-coun...
7         s8  ...  After one of his high school students attacks ...
10       s11  ...  A farmer pens a confession admitting to his wi...
14       s15  ...  Stranded when the Earth is suddenly destroyed ...
...      ...  ...                                                ...
7758   s7759  ...  Zack and Miri make and star in an adult film t...
7771   s7772  ...  Born without legs and stuck in foster care for...
7774   s7775  ...  A political cartoonist, a crime reporter and a...
7778   s7779  ...  Looking to survive in a world taken over by zo...
7781   s7782  ...  Dragged from civilian life, a former superhero...

[2555 rows x 10 columns]

Con este código clasificamos los datos para que solo nos muestre las películas de Estados Unidos.

In [ ]:
India = base[base['country']=='India']

India.to_csv('Solo_India.csv', index = False)
India

show_id  ...                                        description
8         s9  ...  When a doctor goes missing, his psychiatrist w...
9        s10  ...  An architect and his wife move into a castle t...
18       s19  ...  On India's Independence Day, a zany mishap in ...
20       s21  ...  After accidentally connecting over the Interne...
21       s22  ...  A taxi driver, a gangster and an athlete strug...
...      ...  ...                                                ...
7764   s7765  ...  Through his relationships with two wildly diff...
7769   s7770  ...  Three friends on an adventurous road trip/bach...
7776   s7777  ...  When his cruel uncle abandons him, a young orp...
7780   s7781  ...  A drug dealer starts having doubts about his t...
7783   s7784  ...  A scrappy but poor boy worms his way into a ty...

[923 rows x 12 columns]

Con este código clasificamos los datos para que solo nos muestre las películas de India.

**¿Para que me puede servir estas agrupaciones para la práctica siguiente?**
Las principales industrias cinematográficas del mundo son Hollywood y Bollywood; hacer las categorías de Estados Unidos e India nos ayuda a marcar esta división y poder analizar los datos más eficientemente.

In [ ]:
#e)Cambiar nombres de la columnas
base.rename(columns = {'show_id':'id_show'})

id_show  ...                                        description
0         s1  ...  In a future where the elite inhabit an island ...
1         s2  ...  After a devastating earthquake hits Mexico Cit...
2         s3  ...  When an army recruit is found dead, his fellow...
3         s4  ...  In a postapocalyptic world, rag-doll robots hi...
4         s5  ...  A brilliant group of students become card-coun...
...      ...  ...                                                ...
7782   s7783  ...  When Lebanon's Civil War deprives Zozo of his ...
7783   s7784  ...  A scrappy but poor boy worms his way into a ty...
7784   s7785  ...  In this documentary, South African rapper Nast...
7785   s7786  ...  Dessert wizard Adriano Zumbo looks for the nex...
7786   s7787  ...  This documentary delves into the mystique behi...

[7787 rows x 12 columns]

Con esta funcion podemos cambiar el nombre de cualquier columna.

In [ ]:
#f)Guardar archivos en agrupaciones
base.type.unique()

array(['TV Show', 'Movie'], dtype=object)

Esta funcion nos da los valores unicos de cierta columna y asi podremos agruparlos por categorias.

In [ ]:
TV_Show = base[base['type']=='TV Show']

TV_Show.to_csv('Solo_TV_Show.csv', index=False)
TV_Show

show_id  ...                                        description
0         s1  ...  In a future where the elite inhabit an island ...
5         s6  ...  A genetics professor experiments with a treatm...
11       s12  ...  In this dark alt-history thriller, a naïve law...
12       s13  ...  Archival video and new interviews examine Mexi...
16       s17  ...  As a psychology professor faces Alzheimer's, h...
...      ...  ...                                                ...
7767   s7768  ...  Strong-willed, middle-class Kashaf and carefre...
7775   s7776  ...  A quest for freedom and legendary treasure beg...
7777   s7778  ...  While living alone in a spooky town, a young g...
7779   s7780  ...  An assortment of talent takes the stage for a ...
7785   s7786  ...  Dessert wizard Adriano Zumbo looks for the nex...

[2410 rows x 12 columns]

Con esta funcion estamos agrupando todos los datos que en la columna "type" tiene asignada la categoria "TV_Show". Tambien nos muestra cuantas filas tenemos en nuestra nueva base.

In [ ]:
Movie= base[base['type']=='Movie']

Movie.to_csv('Solo_Movie.csv', index=False)
Movie

show_id  ...                                        description
1         s2  ...  After a devastating earthquake hits Mexico Cit...
2         s3  ...  When an army recruit is found dead, his fellow...
3         s4  ...  In a postapocalyptic world, rag-doll robots hi...
4         s5  ...  A brilliant group of students become card-coun...
6         s7  ...  After an awful accident, a couple admitted to ...
...      ...  ...                                                ...
7781   s7782  ...  Dragged from civilian life, a former superhero...
7782   s7783  ...  When Lebanon's Civil War deprives Zozo of his ...
7783   s7784  ...  A scrappy but poor boy worms his way into a ty...
7784   s7785  ...  In this documentary, South African rapper Nast...
7786   s7787  ...  This documentary delves into the mystique behi...

[5377 rows x 12 columns]

Con esta funcion estamos agrupando todos los datos que en la columna "type" tiene asignada la categoria "Movie". Tambien nos muestra cuantas filas tenemos en nuestra nueva base.

**TAREAS EXTRAS**

* **Comando df.dropna:** El comando dropna permite, filtrar los valores de una estructura de datos para dejar solo aquellos no nulos. 


In [ ]:
base.dropna()

show_id  ...                                        description
1         s2  ...  After a devastating earthquake hits Mexico Cit...
2         s3  ...  When an army recruit is found dead, his fellow...
3         s4  ...  In a postapocalyptic world, rag-doll robots hi...
4         s5  ...  A brilliant group of students become card-coun...
5         s6  ...  A genetics professor experiments with a treatm...
...      ...  ...                                                ...
7778   s7779  ...  Looking to survive in a world taken over by zo...
7780   s7781  ...  A drug dealer starts having doubts about his t...
7781   s7782  ...  Dragged from civilian life, a former superhero...
7782   s7783  ...  When Lebanon's Civil War deprives Zozo of his ...
7783   s7784  ...  A scrappy but poor boy worms his way into a ty...

[4808 rows x 12 columns]

Aqui arriba podemos ver que de 7787 filas que tenia nuestra base original solo quedaron los valores no nulos que son 4808 filas.

* **Comando df.astype:** Con este comando es posible hacer un copia de la base aplicándole otro tipo de dato.

In [ ]:
base.astype(bool)

show_id  type  title  director  ...  rating  duration  listed_in  description
0        True  True   True      True  ...    True      True       True         True
1        True  True   True      True  ...    True      True       True         True
2        True  True   True      True  ...    True      True       True         True
3        True  True   True      True  ...    True      True       True         True
4        True  True   True      True  ...    True      True       True         True
...       ...   ...    ...       ...  ...     ...       ...        ...          ...
7782     True  True   True      True  ...    True      True       True         True
7783     True  True   True      True  ...    True      True       True         True
7784     True  True   True      True  ...    True      True       True         True
7785     True  True   True      True  ...    True      True       True         True
7786     True  True   True      True  ...    True      True       True         True

[7787 rows x 12 columns]

En el caso de arriba cambiamos todos nuestros datos de la base original a datos de tipo booleano. 

* **Realizar una tarea de preprocesamiento**

In [ ]:
base.dropna(thresh=int(base.shape[0] * .9), axis=1)

show_id  ...                                        description
0         s1  ...  In a future where the elite inhabit an island ...
1         s2  ...  After a devastating earthquake hits Mexico Cit...
2         s3  ...  When an army recruit is found dead, his fellow...
3         s4  ...  In a postapocalyptic world, rag-doll robots hi...
4         s5  ...  A brilliant group of students become card-coun...
...      ...  ...                                                ...
7782   s7783  ...  When Lebanon's Civil War deprives Zozo of his ...
7783   s7784  ...  A scrappy but poor boy worms his way into a ty...
7784   s7785  ...  In this documentary, South African rapper Nast...
7785   s7786  ...  Dessert wizard Adriano Zumbo looks for the nex...
7786   s7787  ...  This documentary delves into the mystique behi...

[7787 rows x 11 columns]